src/lib.rs

```diff
@@ -71,15 +71,13 @@ pub mod author {
         pub fn parse(author: &str) -> Result<Author, AuthorParseError> {
             let (name, remainder) = author.split_once('<').ok_or(AuthorParseError::NameFailed)?;
 
-            let email = remainder
-                .strip_suffix('>')
-                .ok_or(AuthorParseError::EmailFailed)?
-                .trim()
-                .into();
-
             Ok(Self {
                 name: name.trim().into(),
-                email,
```

Comment:
Why is this change preferred?

Reply:
It is shorter.  Is a shorter source code with the same meaning and readability such a big problem?

Reply:
I actually think it has less readability, since it makes the initialization of the struct more "complex". The structs initializations can become quite large, if you do it this way 😅 

That does not mean, that I never do it. Like in a function, which basically is only the struct initialization I'd prefer it your way, but here I think it makes it nicer to read.

Reply:
Basically, this would require to also rewrite the `Commit` constructor, as well, which you specified the way I rewrote the `Author` constructor.

Finally, the way you specified the `Commit` constructor was the reason for me to rewrite the `Author` constructor:  to keep the overall appearance of the code uniform.

Reply:
Functional programming is the technique to reduce the count of statements without changing the meaning.  If you are familiar with Haskell and related languages, you consider purely functional source code way more readable than imperative code with the same meaning.  It also reduces the verbosity of the code, this is the set of obsolete assignments just because one single change is made to a variable.

You used mixed verbosity in your original definition, I tried to unify it to the preferable functional style of software engineering.

Reply:
I actually do not think shorter code is more readable, more often than not the contrary is true. I don't know, if you know the `APL` language, people that write it, praise its shortness and readability, but to me it always feels more like solving a puzzle.. 😅 

Yeah, like I said in the other comment, I guess, I have not yet developed a style here.